pkg/app/piped/livestatestore/cloudrun/store.go

```diff
@@ -44,54 +104,68 @@ func (s *store) run(ctx context.Context) error {
 		// https://cloud.google.com/run/quotas#api
 		v, next, err := s.client.List(ctx, ops)
 		if err != nil {
-			return fmt.Errorf("failed to list cloudrun services: %w", err)
+			return nil, err
 		}
-		svc = append(svc, v...)
+		svcs = append(svcs, v...)
 		if next == "" {
 			break
 		}
 		cursor = next
 	}
-
-	// Update apps to the latest.
-	s.setApps(svc)
-
-	return nil
+	return svcs, nil
 }
 
-func (s *store) setApps(svc []*provider.Service) {
-	apps := make(map[string]provider.ServiceManifest, len(svc))
-	for i := range svc {
-		sm, err := svc[i].ServiceManifest()
+func (s *store) getMultiRevisions(ctx context.Context, names []string) []*provider.Revision {
+	ret := make([]*provider.Revision, len(names))
+	for i := range names {
+		v, err := s.client.GetRevision(ctx, names[i])
```

Comment:
This costs many calls. So do we have any way to fetch a list of revisions by the label?

Reply:
The best scenario is that we can fetch all revisions which contain `managed-by: pipecd` label.
I hope that they provide that kind of API. 😄

Reply:
I also thought about how to list it with label, but it will be a tremendous number because it include inactive revision.
It may be called a lot in the above implementation, but I thought it's more efficient because it only gets the active revisions.
The best way is to list revision which contain `managed-by: pipecd` and active if we can do...

Reply:
@knanao Looks like they are having a `list` API to retrieve a list of revisions. Can we use it?

https://cloud.google.com/run/docs/reference/rest/v1/namespaces.revisions/list

Reply:
@nghialv 
I fixed this in https://github.com/pipe-cd/pipecd/pull/3240/commits/f2927cfcfe03709dc3d7297e3d3914b5f833fab3, so PTAL when you can do.